bootstrap_test.py

```diff
@@ -280,6 +280,55 @@ def test_consistent_range_movement_true_with_rf1_should_fail(self):
     def test_consistent_range_movement_false_with_rf1_should_succeed(self):
         self._bootstrap_test_with_replica_down(False, rf=1)
 
+    def test_rf_gt_nodes_multidc_should_succeed(self):
+        """
+        Test creating a KS with RF > N on multi DC doesn't break bootstrap
+        @jira_ticket CASSANDRA-16296 CASSANDRA-16411
+        """
+        cluster = self.cluster
+        cluster.set_environment_variable('CASSANDRA_TOKEN_PREGENERATION_DISABLED', 'True')
+        cluster.populate([1, 1])
+        cluster.start()
+
+        node1 = cluster.nodelist()[0]
+        session = self.patient_cql_connection(node1)
+        session.execute("CREATE KEYSPACE k WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'dc1' : '3'}")
+        warning = 'Your replication factor 3 for keyspace k is higher than the number of nodes 1 for datacenter dc1'
+
+        if cluster.version() >= '4.0':
+            warnings_total = 0
```

Comment:
```
warnings_total = sum([len(node.grep_log(warning) for node in cluster.node_list()])
assert warnings_total == 1
```
but actually what we would like to check is: "warning on a dc1 node, and no warning on dc2 node"
so maybe more explicit:
```
assert len(node1.grep_log(warning)) == 1
assert len(node2.grep_log(warning)) == 0
```

Reply:
Good point, that makes more sense. Just note that the session created with `patient_cql_connection` can use any node as coordinator so doing the asserts that way would be flaky unless we also create the session with `patient_exclusive_cql_connection`. We can also put `warning` inside the `if` block since I think we are not going to reuse it:
```python
node1, node2 = cluster.nodelist()
session = self.patient_exclusive_cql_connection(node1)
session.execute("CREATE KEYSPACE k WITH REPLICATION = {'class' : 'NetworkTopologyStrategy', 'dc1' : '3'}")

if cluster.version() >= '4.0':
    warning = 'Your replication factor 3 for keyspace k is higher than the number of nodes 1 for datacenter dc1'
    assert len(node1.grep_log(warning)) == 1
    assert len(node2.grep_log(warning)) == 0
```

Reply:
Oh I didn't know about the 'exclusive' cql connection gtk.